tx-pool/src/component/container.rs

```diff
@@ -275,10 +275,22 @@ impl SortedTxMap {
         TxLink::get_descendants(&self.links, tx_short_id)
     }
 
-    /// return sorted keys
-    pub fn sorted_keys(&self) -> impl Iterator<Item = &AncestorsScoreSortKey> {
+    /// return keys sorted by tx fee rate
+    pub fn keys_sorted_by_fee(&self) -> impl Iterator<Item = &AncestorsScoreSortKey> {
         self.sorted_index.iter().rev()
     }
+
+    /// sorted keys by tx fee rate and transaction relation
+    pub fn keys_sorted_by_fee_and_relation(&self) -> Vec<&AncestorsScoreSortKey> {
+        let mut sorted: Vec<_> = self.sorted_index.iter().rev().collect();
+        sorted.sort_by_key(|k| {
+            self.entries
+                .get(&k.id)
+                .expect("entries should consistent with sorted_index")
+                .ancestors_count
+        });
```

Comment:
Could we reduce the `rev()` and `collect()`? It seems useless.

```
self.sorted_index.iter()
     .sort_by_key(|k| {
            self.entries
                .get(&k.id)
                .expect("entries should consistent with sorted_index")
                .ancestors_count
        });
```

Reply:
no method named `sort_by_key` found for type `std::iter::Rev<std::collections::btree_set::Iter<'_, component::container::AncestorsScoreSortKey>>`

Reply:
Please just call `keys_sorted_by_fee` method, 
the `rev` is an iterator method, has no performance issue, and collect is nessacery for sorting, please review more carefully.